src/cli.rs

```diff
@@ -139,8 +184,10 @@ pub struct Config {
     )]
     pub chain_update_interval: Duration,
 
-    // Dataset alias -> bucket URL
-    pub available_datasets: HashMap<String, String>,
+    #[serde(rename = "serve", default = "default_served_datasets")]
+    pub available_datasets: Vec<DatasetConfig>,
```

Comment:
Can we please separate the static and updatable parts of the config? It will be much easier in the code if the `Config` is a read-only struct initialized by parsing the file once and the `DatasetsConfig` is some concurrently accessed mapping that is passed with a separate axum extension. You can make some methods of the `Config` accept a reference to the `DatasetsConfig` or vice versa and only "glue" them on the calling side

Reply:
Okay.